x-pack/elastic-agent/pkg/agent/transpiler/rules_test.go

```diff
@@ -22,6 +22,36 @@ func TestRules(t *testing.T) {
 		expectedYAML string
 		rule         Rule
 	}{
+		"overwrite": {
+			givenYAML: `
+fleet:
+  kibana_url: contents
+output:
+  elasticsearch:
+    hosts: []
+datasources:
+  - name: All default
+    inputs:
+    - type: file
+      streams:
+        - paths: /var/log/mysql/error.log
+`,
+			expectedYAML: `
+type: file
```

Comment:
I stumbled over this, is the below correct or should it be inside inputs? Because if we put all the levels just together, keys might conflict.